### PR TITLE
Fix race condition

### PR DESCRIPTION
**Pull request by nickpell**
http://manage.dimagi.com/default.asp?189979 is reproducible by running multiple `rebuild_export_task` concurrently.

Cleanly aborts losing tasks.

@czue @orangejenny @TylerSheffels

### CODE REVIEW

**Comment by czue** [comment]
interesting. do we know why these would get run concurrently?

File: corehq/ex-submodules/couchexport/groupexports.py [hunk 1]
```diff
@@ -54,7 +54,11 @@ def rebuild_export(config, schema, output_dir, last_access_cutoff=None, filter=N
             if saved.last_accessed is None:
                 saved.last_accessed = datetime.utcnow()
             saved.last_updated = datetime.utcnow()
-            saved.save()
+            try:
+                saved.save()
+            except ResourceConflict:
+                # task was executed concurrently, so let first to finish win and abort the rest
+                return
```
Comment (czue) [inline corehq/ex-submodules/couchexport/groupexports.py]: (not blocking but using `try`/`else` might make the control flow here a little cleaner)

**Comment by czue** [comment]
ok to merge when tests pass, though i'm curious about root cause of multiple of these getting fired.

**Comment by nickpell** [comment]
@czue Can be fired from the UI: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/export/views.py#L1008

File: corehq/ex-submodules/couchexport/groupexports.py (hunk 1 above)
Comment (nickpell) [inline corehq/ex-submodules/couchexport/groupexports.py]: like this?

```
try:
    saved.save()
except ResourceConflict:
    pass
else:
    saved.set_payload(payload)
```

**Comment by czue** [comment]
interesting. still isn't a super satisfying answer to the increase in errors recently but could have just been an odd coincidence

**Pull request merged**